src/ducks/connections/components/queue/queue.jsx

```diff
@@ -10,30 +10,64 @@ const Pending = translate()(props => (
   </span>
 ))
 
-const Item = translate()(({ t, label, status, icon }) => {
-  return (
-    <div
-      className={classNames(styles['queue-item'], {
-        [styles['queue-item--done']]: status === 'done',
-        [styles['queue-item--error']]: status === 'error'
-      })}
-    >
-      <div className={classNames(styles['item-icon'])}>
-        {icon && (
-          <img className={classNames(styles['item-icon-img'])} src={icon} />
-        )}
-      </div>
-      <div className={classNames(styles['item-label'])}>{label}</div>
-      <div className={styles['item-status']}>
-        {status === 'pending' ? (
-          <Pending />
-        ) : (
-          <div className={styles[`item-${status}`]} />
-        )}
+class Item extends Component {
+  state = {
+    progress: 0
+  }
+  componentDidMount() {
+    let elapsedTime = 0
+    this.myInterval = setInterval(() => {
+      elapsedTime += 10
+      let progress = Math.atan(elapsedTime / 3e3) / (Math.PI / 2) * 90
+      this.setState({
+        progress: progress
+      })
+    }, 25)
+  }
+  componentDidUpdate = prevProps => {
+    const { progress } = this.state
+    progress >= 100 && clearInterval(this.myInterval)
```

Comment:
Is this line actually used? For me the `atan()` function will never reach the threshold (which is 90 here), so we could remove it since it will never happen.